benches/benches.rs

```diff
@@ -0,0 +1,149 @@
+    });
+    c.bench_function("Cached AOT fib_2M", |b| {
+        b.iter(|| aot_fibonacci.invoke_dynamic(fibonacci_function_id, &[], Some(u128::MAX), None));
+    });
+    c.bench_function("Cached AOT logistic_map", |b| {
+        b.iter(|| {
+            aot_logistic_map.invoke_dynamic(logistic_map_function_id, &[], Some(u128::MAX), None)
+        });
+    });
+
+    unsafe {
+        #[allow(dead_code)]
+        struct PanicResult {
+            tag: u8,
+            payload: MaybeUninit<(i32, i32, *mut [u64; 4])>,
+        }
+
+        let aot_factorial_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                aot_factorial
+                    .find_function_ptr(factorial_function_id)
+                    .cast(),
+            );
+        let aot_fibonacci_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                aot_fibonacci
+                    .find_function_ptr(fibonacci_function_id)
+                    .cast(),
+            );
+        let aot_logistic_map_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                aot_logistic_map
+                    .find_function_ptr(logistic_map_function_id)
+                    .cast(),
+            );
+        let jit_factorial_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                jit_factorial
+                    .find_function_ptr(factorial_function_id)
+                    .cast(),
+            );
+        let jit_fibonacci_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                jit_fibonacci
+                    .find_function_ptr(fibonacci_function_id)
+                    .cast(),
+            );
+        let jit_logistic_map_fn =
+            std::mem::transmute::<*const (), extern "C" fn(u128) -> (u128, PanicResult)>(
+                jit_logistic_map
+                    .find_function_ptr(logistic_map_function_id)
+                    .cast(),
+            );
+
+        c.bench_function("Cached JIT factorial_2M (direct invoke)", |b| {
+            b.iter(|| jit_factorial_fn(u128::MAX));
+        });
+        c.bench_function("Cached JIT fib_2M (direct invoke)", |b| {
+            b.iter(|| jit_fibonacci_fn(u128::MAX));
+        });
+        c.bench_function("Cached JIT logistic_map (direct invoke)", |b| {
+            b.iter(|| jit_logistic_map_fn(u128::MAX));
+        });
+
+        c.bench_function("Cached AOT factorial_2M (direct invoke)", |b| {
+            b.iter(|| aot_factorial_fn(u128::MAX));
+        });
+        c.bench_function("Cached AOT fib_2M (direct invoke)", |b| {
+            b.iter(|| aot_fibonacci_fn(u128::MAX));
+        });
+        c.bench_function("Cached AOT logistic_map (direct invoke)", |b| {
+            b.iter(|| aot_logistic_map_fn(u128::MAX));
+        });
```

Comment:
Does this part need to be in the `unsafe` block?

Reply:
It was necessary on some of my first attempts, but not anymore. Fixed.